projects/packages/jetpack-mu-wpcom/src/features/wpcom-comments/wpcom-comments.js

```diff
@@ -0,0 +1,46 @@
+document.addEventListener( 'DOMContentLoaded', () => {
+	const likeLinks = [ ...document.getElementsByClassName( 'comment-like-link' ) ];
+
+	likeLinks.forEach( link => {
+		link.addEventListener( 'click', event => {
+			event.preventDefault();
+
+			const span = link.querySelector( 'span' );
+			const currentText = span.textContent.trim();
+			const currentAction = currentText === 'Like' ? 'like_comment' : 'unlike_comment';
+
+			// Prepare the data payload
+			const postData = new URLSearchParams( {
+				action: currentAction,
+				_wpnonce: link.dataset.nonce,
+				like_comment: link.dataset.comment,
+				blog_id: link.dataset.blog,
+			} );
+
+			// Send the request
+			fetch( '/wp-admin/admin-ajax.php', {
```

Comment:
This question speaks to my ignorance of WPCOM coming from Dotorg, but:

As far as I can tell ([OpenGrok](https://opengrok.a8c.com/source/search?full=wp_ajax_like_comment&defs=&refs=&path=&hist=&type=&xrd=&nn=16&searchall=true)), the AJAX handlers for `like_comment` and `unlike_comment` are defined in two different places:
* /mu-plugins/likes/
* /mu-plugins/comment-likes/

while this particular implementation lives in `jetpack-mu-wpcom`. Is this separation standard practice? Since there are already two locations where comment-like functionality exists, how does a developer weigh where to implement this PR's change, and why adding a third location?

Reply:
Hi, this is still in progress so don't pay too much attention to the actual code and implementation. I'm still in a phase where I'm just getting things to work. It's best to wait for the PR to be ready for review since the implementation isn't final and is missing layers of polish XD.

My local changes are drastically different from the current changes on this PR.

Reply:
>  Is this separation standard practice? Since there are already two locations where comment-like functionality exists, how does a developer weigh where to implement this PR's change, and why adding a third location?

There isn't a standard practice when it comes to things like this. We have lots of legacy code and most of the time we need to figure things out on the go. In this case, I'm 90% confident that the final implementation will rely on `public-api.wordpress.com/rest/v1.2/sites/$blog_id/posts/$post_id/likes`, for retrieving, liking, and unblinking.

This functionality needs to work on simple and Atomic sites so we can't rely on wpcom's mu-plugins directly if that makes sense.

Reply:
Fair enough, and sorry if I got ahead of things. :) A part of this is curiosity to learn. Hit me up when the PR is ready!

Reply:
> Fair enough, and sorry if I got ahead of things. :) A part of this is curiosity to learn. Hit me up when the PR is ready!

No problem, I'll ping you once it's ready for review.